trie-db/src/triedbmut.rs

```diff
@@ -711,18 +742,144 @@ impl<'db, L: TrieLayout> TrieDBMutBuilder<'db, L> {
 			hash_count: 0,
 			storage: NodeStorage::empty(),
 			death_row: Default::default(),
+			death_row_child: Default::default(),
 			root_handle,
 		}
 	}
 }
 
+#[derive(Debug)]
+pub struct NewChangesetNode<H, DL> {
+	pub hash: H,
+	pub prefix: OwnedPrefix,
+	pub data: Vec<u8>,
+	pub children: Vec<ChangesetNodeRef<H, DL>>,
+	// Storing the key and removed nodes (only needed for old trie).
+	pub key_childset: Option<(Vec<u8>, Vec<(H, OwnedPrefix)>)>,
```

Comment:
change from arkpar/new-trie for substrate child trie usecase support.
Not that this is not used when using locations, but I don't think it is worth making conditional.
On the other hand not feeding the removed nodes and debug asserting it is empty when using location may be a good idea.